### PR TITLE
✨ Create Blip

### PR DESCRIPTION
**Pull request by Walker974**
Hi , I made sure to create blip in server side .
I did'nt find any other solution.

### CODE REVIEW

File: src/api/blip/server/createBlip.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- create Blip
+---@param blip_name string
+---@param blip_coords table
+---@param blip_sprite number
+---@param blip_display number
+---@param blip_color number
+---@param blip_scale number
+---@public
```
Comment (MoskalykA) [inline src/api/blip/server/createBlip.lua:8]:
```suggestion
---create Blip
---@param blip_name string
---@param blip_coords table
---@param blip_sprite number
---@param blip_display number
---@param blip_color number
---@param blip_scale number
---@return void
---@public
```

File: src/api/blip/client/customBlip.lua [hunk 2]
```diff
@@ -0,0 +1,13 @@
+_NCS:registerNetEvent('API_Blip:customBlip')
+_NCS:handleEvent('API_Blip:customBlip', function(blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
+    Blip = AddBlipForCoord(blip_coords.x, blip_coords.y, blip_coords.z)
```
Comment (MoskalykA) [inline src/api/blip/client/customBlip.lua:3]:
```suggestion
    local Blip = AddBlipForCoord(blip_coords.x, blip_coords.y, blip_coords.z)
```

**Review by MoskalykA — CHANGES_REQUESTED** [review]

**Comment by Walker974** [comment]
good now ?

File: src/api/blip/client/customBlip.lua [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    SetBlipSprite(Blip, blip_sprite)
+    SetBlipDisplay(Blip, blip_display)
+    SetBlipScale(Blip, blip_scale)
+    SetBlipColour(Blip, blip_color)
+
+    SetBlipAsShortRange(Blip, true)
+    BeginTextCommandSetBlipName("STRING")
+    AddTextComponentString(blip_name)
+    EndTextCommandSetBlipName(Blip)
+end)
```
Comment (MoskalykA) [inline src/api/blip/client/customBlip.lua:14]:
```suggestion
_NCS:registerNetEvent('API_Blip:customBlip')
_NCS:handleEvent('API_Blip:customBlip', function(blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
    local Blip = AddBlipForCoord(blip_coords.x, blip_coords.y, blip_coords.z)
    SetBlipSprite(Blip, blip_sprite)
    SetBlipDisplay(Blip, blip_display)
    SetBlipScale(Blip, blip_scale)
    SetBlipColour(Blip, blip_color)

    SetBlipAsShortRange(Blip, true)
    BeginTextCommandSetBlipName("STRING")
    AddTextComponentString(blip_name)
    EndTextCommandSetBlipName(Blip)
end)
```

File: src/api/blip/server/createBlip.lua [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+---@param blip_sprite number
+---@param blip_display number
+---@param blip_color number
+---@param blip_scale number
+---@return void
+---@public
+
+function API_Blip:createBlip(blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
+    _NCS:triggerClientEvent('API_Blip:customBlip', -1, blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
+end
```
Comment (MoskalykA) [inline src/api/blip/server/createBlip.lua:13]:
```suggestion
---createBlip
---@param blip_name string
---@param blip_coords table
---@param blip_sprite number
---@param blip_display number
---@param blip_color number
---@param blip_scale number
---@return void
---@public
function API_Blip:createBlip(blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
    _NCS:triggerClientEvent('API_Blip:customBlip', -1, blip_name, blip_coords, blip_sprite, blip_display, blip_scale, blip_color)
end
```

**Review by MoskalykA — CHANGES_REQUESTED** [review]

**Comment by Walker974** [comment]
good ?

**Review by MoskalykA — APPROVED** [review]

**Comment by Wisfakim** [comment]
Maybe you can add blip management list at client side ? In case if you want remove blip ? ( exemple for a job)

**Comment by MoskalykA** [comment]
> Maybe you can add blip management list at client side ? In case if you want remove blip ? ( exemple for a job)

It would be enough to make a global variable on the client side and to assign the blips to this variable with an id. After that, you would have to make a function that takes the id as a parameter and deletes the blip from the table.
@Walker974 you want to do this feature?

**Comment by MoskalykA** [comment]
> Maybe you can add blip management list at client side ? In case if you want remove blip ? ( exemple for a job)

It would be enough to make a global variable on the client side and to assign the blips to this variable with an id. After that, you would have to make a function that takes the id as a parameter and deletes the blip from the table.

**Comment by MoskalykA** [comment]
Sorry but Pablo is going to import one of his systems that will handle zones, blips and npcs.
Thanks for your contribution and for the future try to make less commits but also use the gitmoji conventions. (https://gitmoji.dev/)

**Pull request closed without merging**